google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -90,6 +94,16 @@ private void checkSdk() {
       return;
     }
 
+    if (System.getProperty("os.name").contains("Windows")
+        && !path.matches(WIN_PATH_REGEX)) {
```

Comment:
Rather than going this route, did you consider just calling Paths.get in a try...catch block, and catching the `InvalidPathException`? One advantage of going that way is you don't need to perform any special regex matching or platform-specific validation in our codebase.